homeassistant/components/ecovacs_mqtt/const.py

```diff
@@ -0,0 +1,29 @@
+"""Const module."""
+from __future__ import annotations
+
+from enum import StrEnum
+
+from homeassistant.const import (
+    CONF_COUNTRY,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+
+DOMAIN = "ecovacs_mqtt"
+
+
+class Mode(StrEnum):
+    """Enum for modes."""
+
+    CLOUD = "cloud"
+    SELF_HOSTED = "self_hosted"
+
+
+# Self hosted (bumper) has no auth and serves the urls for all countries/continents
+SELF_HOSTED_CONFIGURATION = {
+    CONF_COUNTRY: "it",
+    CONF_PASSWORD: "password",
+    CONF_USERNAME: "username",
+    CONF_VERIFY_SSL: False,  # required it is using self signed certificates
```

Comment:
Why is this config hardcoded? What does this do?

Reply:
[Bumper](https://github.com/bmartin5692/bumper) is a project to simulate the ecovacs server (mqtt broker and rest to mqtt converter) so the bot is not communicating with any cloud server. This project uses self-signed certificates to create certs for ecovacs domains.
It will be set to False only for self hosted installations.
Is this fine or should I still ask the user for it? (Only on self hosted installations)

Reply:
Why is the config needed at all then?

Reply:
Because bumper accepts any auth information (it does not implement any auth), asking the user for a username and password will be useless.  To avoid it I use this config and skip the `auth` step in the config flow

Reply:
If Bumper doesn't implement any auth, why do we need to pass auth config? Shouldn't that be optional in that case depending on what mode is used?

Side note: We currently don't skip the auth step in the config flow, but pass this hardcoded config into that step.